src/MultiRunner.php

```diff
@@ -84,18 +89,27 @@ public function remove(PromiseCore $core): void
     public function wait(PromiseCore $targetCore = null): void
     {
         do {
-            $status = curl_multi_exec($this->multiHandle, $active);
+            if (curl_multi_select($this->multiHandle, self::SELECT_TIMEOUT) === -1) {
+                // See https://bugs.php.net/bug.php?id=61141
+                usleep(250);
+            }
+
+            do {
+                $status = curl_multi_exec($this->multiHandle, $active);
```

Comment:
is this now again busy waiting? should we repeat the curl_multi_select in each loop?

Reply:
I'm not pretty sure. Most examples I've found are like this https://bugs.php.net/bug.php?id=61141#1348321490

Reply:
its again the same as guzzle does, which is most likely what the symfony http client is using, so i'd hope its the right thing.